File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/tv/series/model/TvSeriesNetwork.java
package dev.reuss.tmdb.domain.tv.series.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;

/**
 * TV network used in TV series details.
 *
 * @param id TMDB network id
 * @param logoPath logo image path
 * @param name network name
 * @param originCountry origin country code
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvSeriesNetwork(
        int id,

        @JsonProperty("logo_path")
        String logoPath,

        String name,

        @JsonProperty("origin_country")
        String originCountry
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/tv/series/model/TvSeriesNetwork.kt
package dev.reuss.tmdb.domain.tv.series.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * TV network used in TV series details.
 *
 * @property id TMDB network id
 * @property logoPath logo image path
 * @property name network name
 * @property originCountry origin country code
 */
@JvmRecord
data class TvSeriesNetwork(
    @all:JsonProperty("id")
    val id: Int,

    @all:JsonProperty("logo_path")
    val logoPath: String?,

    @all:JsonProperty("name")
    val name: String?,

    @all:JsonProperty("origin_country")
    val originCountry: String?
) : TmdbModel